entities/library.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class Library
+  include Validation
+  include Statistics
+
+  attr_accessor :authors, :readers, :books, :orders
+
+  def initialize
+    @authors = []
+    @books = []
+    @readers = []
+    @orders = []
+
+    load_datas
+  end
+
+  def load_datas
+    @authors = YAML.load_file(AUTHORS_DB) if File.exist?(AUTHORS_DB)
+    @readers = YAML.load_file(READERS_DB) if File.exist?(READERS_DB)
+    @books = YAML.load_file(BOOKS_DB) if File.exist?(BOOKS_DB)
+    @orders = YAML.load_file(ORDERS_DB) if File.exist?(ORDERS_DB)
+  end
+
+  def save
+    File.open(AUTHORS_DB, 'w') { |file| file.write(@authors.to_yaml) }
+    File.open(READERS_DB, 'w') { |file| file.write(@readers.to_yaml) }
+    File.open(BOOKS_DB, 'w') { |file| file.write(@books.to_yaml) }
+    File.open(ORDERS_DB, 'w') { |file| file.write(@orders.to_yaml) }
+  end
+
+  def add_entity(entities_input:, arr:)
+    found = nil
+    arr.each { |entities| found = entities if entities == entities_input }
+    found || arr.push(entities_input) && entities_input
+  end
+
+  def add_author(author_input)
```

Comment:
lets define one method **add_entity** which will add Book or Author or Reader or Order to library and will raise an exception if passed entity isn't one of above classes.

Reply:
this will entail additional data processing (each entity has its own array)